mcs/class/System/System.Net.Sockets/SocketAsyncResult.cs

```diff
@@ -153,9 +158,8 @@ public void Complete ()
 			Socket completedSocket = socket;
 			SocketOperation completedOperation = operation;
 
-			AsyncCallback callback = AsyncCallback;
-			if (callback != null) {
-				ThreadPool.UnsafeQueueUserWorkItem (_ => callback (this), null);
+			if (this.AsyncCallback != null) {
```

Comment:
This is not racy because `AsyncCallback` is readonly and initialised at object construction.